src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -52,15 +60,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     openHours.add(Integer.parseInt(hourOpen));
     openHours.add(Integer.parseInt(hourClosed));
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String username;
+    if (isUserLoggedIn) {
+      /* Currently uses user email to be consistent w other parts of codebase, subject to change */
```

Comment:
Normally I'm all for consistency, but when I'd be extra careful about the cost of a bug in history tracking. What's the downside of moving to the correct id system now?

Reply:
I wrote that it's subject to change if we decide to change to tracking users by userId instead of email, but right now it's using the emails to ensure that the history entries can show who wrote it without needing to do a conversion on the userid property

Reply:
Though the other parts of the code will change to pivot around userId, I think it's OK to break consistency here as long as you are tracking the change elsewhere.